packages/kbn-config/src/deprecation/apply_deprecations.ts

```diff
@@ -22,9 +23,21 @@ export const applyDeprecations = (
   deprecations: ConfigDeprecationWithContext[],
   createAddDeprecation: (pluginId: string) => AddConfigDeprecation = noopAddDeprecationFactory
 ) => {
-  let processed = cloneDeep(config);
+  const result = cloneDeep(config);
   deprecations.forEach(({ deprecation, path }) => {
-    processed = deprecation(processed, path, createAddDeprecation(path));
+    const commands = deprecation(result, path, createAddDeprecation(path));
```

Comment:
@Bamieh In a follow-up, we can either store all the commands to report them later.  Do we need original values? Or config keys only?

Reply:
@mshustov for the telemetry collector we'd only need the deprecated config keys